llvm/tools/sycl-post-link/sycl-post-link.cpp

```diff
@@ -271,6 +276,21 @@ static void collectKernelModuleMap(
   }
 }
 
+// Gets reqd_work_group_size information for function Func.
+static std::vector<uint32_t>
+getKernelReqdWorkGroupSizeMetadata(const Function &Func) {
+  auto ReqdWorkGroupSizeMD = Func.getMetadata("reqd_work_group_size");
+  if (!ReqdWorkGroupSizeMD || ReqdWorkGroupSizeMD->getNumOperands() != 3)
```

Comment:
Note: per SYCL 2020 spec, this attribute can have 1 or 2 arguments and according to KhronosGroup/SYCL-Docs#142 there is no default value defined for "missing" arguments, because those versions are considered to be a separate attributes.

Looking at the codebase, it seems like we still always generate metadata with values for all three dimensions, but I wonder what will happen in longer term? (#3743 is probably related as well)

Do we need some kind of `TODO` here?

Reply:
Good point! I wonder, would an assert be too aggressive here? If I make the `getNumOperands` check an assert it should fail as any such changes are introduced, rather than it just not adding the new properties. As it is right now, I suppose the bugs introduced by having fewer than 3 operands likely wouldn't be detected immediately.

Reply:
> I wonder, would an assert be too aggressive here? 

I actually think that it should be fine - it will indeed help spot the attribute codegen change

Reply:
I have added both. Assert on the sizes, with a TODO saying to relax this assumption when the metadata is more relaxed.